ec/ecresource/extensionresource/delete.go

```diff
@@ -0,0 +1,48 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package extensionresource
+
+import (
+	"context"
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/apierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func deleteResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var client = meta.(*api.API)
+
+	_, err := deleteRequest(client, d)
+
+	if err != nil {
+		if alreadyDestroyed(err) {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func alreadyDestroyed(err error) bool {
+	// If the extension is already destroyed, API return 403.
+	return apierror.IsRuntimeStatusCode(err, 403)
```

Comment:
I'm not convinced the API would ever return a `403` this is probably from the c/p. The error check for `extensions.DeleteExtensionNotFound` is missing here. It needs to be checked with `errors.As`.

```suggestion
	var extensionNotFound * extensions.DeleteExtensionNotFound
	return errors.As(err, &extensionNotFound)
```

Reply:
fixed by 61bdf4c

Reply:
fixed by 89f1147